metricflow/naming/linkable_spec_name.py

```diff
@@ -44,18 +47,30 @@ def from_name(qualified_name: str) -> StructuredLinkableSpecName:
         if associated_granularity:
             #  e.g. "ds__month"
             if len(name_parts) == 2:
-                return StructuredLinkableSpecName((), name_parts[0], associated_granularity)
+                return StructuredLinkableSpecName(
+                    entity_link_names=(), element_name=name_parts[0], time_granularity=associated_granularity
+                )
             # e.g. "messages__ds__month"
-            return StructuredLinkableSpecName(tuple(name_parts[:-2]), name_parts[-2], associated_granularity)
+            return StructuredLinkableSpecName(
+                entity_link_names=tuple(name_parts[:-2]),
+                element_name=name_parts[-2],
+                time_granularity=associated_granularity,
+            )
+
         # e.g. "messages__ds"
         else:
-            return StructuredLinkableSpecName(tuple(name_parts[:-1]), name_parts[-1])
+            return StructuredLinkableSpecName(entity_link_names=tuple(name_parts[:-1]), element_name=name_parts[-1])
 
     @property
     def qualified_name(self) -> str:
-        """Return the full name form. e.g. ds or listing__ds__month."""
+        """Return the full name form. e.g. ds or listing__ds__month.
+
+        If date_part is specified, don't include granularity in qualified_name since it will not impact the result.
+        """
         items = list(self.entity_link_names) + [self.element_name]
-        if self.time_granularity:
+        if self.date_part:
+            items.append(self.date_part_suffix(date_part=self.date_part))
+        elif self.time_granularity:
```

Comment:
> does it blow up with a TypeError

It won't blow up with a `TypeError` - it should parse to something like `ds__extract_year`, and it will think that's the `entity__dimension` name. Then the user will get an error because that dimension doesn't exist.

> does it work for any values that happen to be the same across the two enums?

The values won't match granularity values because all the date part column names include `extract_` (no dunder).

Initially I added support for parsing `date_part` in the `from_name` function, but then decided to remove it because I know we're trying to move away from using the dunder syntax for anything except `entity__dimension`!